gossip/scrambler_new/tx_scrambler_test.go

```diff
@@ -0,0 +1,544 @@
+package scrambler
+
+import (
+	"cmp"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"math/rand/v2"
+	"slices"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+// dummyScramblerEntry represents scramblery entry data used for testing
+type dummyScramblerEntry struct {
+	hash     common.Hash    // transaction hash
+	sender   common.Address // sender of the transaction
+	nonce    uint64         // transaction nonce
+	gasPrice *big.Int       // transaction gasPrice
```

Comment:
nit: you could simplify your test code by making this an integer and produce the "big.Int" in the `GetPrice` function.